core/shared/src/main/scala/zio/internal/FiberContext.scala

```diff
@@ -243,6 +243,7 @@ private[zio] final class FiberContext[E, A](
         var opcount: Int = 0
 
         while (curZio ne null) {
+          while (curZio.tag == ZIO.Tags.Resume) { curZio = nextInstr(()) }
```

Comment:
It looks like `nextInstr` can return `null` which could cause `curZio.tag` to throw a `NullPointerException`?

Reply:
I've added a guard condition.